### PR TITLE
PostContingencyResult status fix

### PR DESCRIPTION
**Pull request by Lisrte**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

`PostContingencyResult` status is set to `CONVERGED` if a result file is found, but the proper status is never parsed.

**What is the new behavior (if this is a feature change)?**
Parse `aggregatedResults.xml` and use it for `PostContingencyResult` status

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

### CODE REVIEW

File: dynaflow/src/main/java/com/powsybl/dynaflow/results/FailedCriterion.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ */
+package com.powsybl.dynaflow.results;
+
+/**
+ * scenarioResults or loadIncreaseResults failed criterion
+ * @param message Failed criterion message
+ * @param time Failure time (in seconds)
+ * @author Laurent Issertial {@literal <laurent.issertial at rte-france.com>}
+ */
+public record FailedCriterion(String message, double time) {
```
Comment (flo-dup) [inline dynaflow/src/main/java/com/powsybl/dynaflow/results/FailedCriterion.java:16]: It looks like a simple timestamped message so it's a bit strange to have this FailedCriterion name, don't you think? maybe renaming message into failureDescription, and failureTime?

Comment (Lisrte) [reply dynaflow/src/main/java/com/powsybl/dynaflow/results/FailedCriterion.java:16]: This record is made from the `criterionNonRespected` element in `aggregatedResults.xml` targeting specifically failure form criteria given to the simulation via `criteria.crt` input file. We can rename the record but I think we should keep the _criterion_ keyword.

File: dynaflow/src/main/java/com/powsybl/dynaflow/results/ContingencyResultsUtils.java [hunk 2]
```diff
@@ -48,22 +54,29 @@ public static PreContingencyResult getPreContingencyResult(Network network, Limi
...
+        Map<String, Status> scenarioResults = new HashMap<>();
+        if (Files.exists(results)) {
+            new XmlScenarioResultParser().parse(results, s -> scenarioResults.put(s.id(), s.status()));
+        }
         return contingencies.stream()
-                .map(c -> getPostContingencyResult(network, violationFilter, constraintsDir, c))
+                .map(c -> new PostContingencyResult(c,
+                        ResultsUtil.convertStatus(scenarioResults.getOrDefault(c.getId(), Status.EXECUTION_PROBLEM)),
+                        getLimitViolationsResult(network, violationFilter, constraintsDir, c)))
                 .collect(Collectors.toList());
```
Comment (flo-dup) [inline dynaflow/src/main/java/com/powsybl/dynaflow/results/ContingencyResultsUtils.java:68]:
```suggestion
                .toList();
```

File: dynaflow/src/main/java/com/powsybl/dynaflow/results/FailedCriterion.java (hunk 1 above)
Comment (flo-dup) [reply dynaflow/src/main/java/com/powsybl/dynaflow/results/FailedCriterion.java:16]: Ok, could you still rename message into description?

**Review by flo-dup — APPROVED** [review]

**Pull request merged**